pkg/apis/core/v1/validation/validation.go

```diff
@@ -144,6 +158,89 @@ func ValidatePodLogOptions(opts *v1.PodLogOptions) field.ErrorList {
 	return allErrs
 }
 
+// ValidateNodeLogViewerOptions checks if options that are set are at the correct
+// value. Any incorrect value will be returned to the ErrorList.
+func ValidateNodeLogViewerOptions(opts *v1.NodeLogViewerOptions) field.ErrorList {
+	files, allErrs := validateQuery(opts.Query)
+	if files && (opts.SinceTime != nil || opts.UntilTime != nil || opts.Boot != nil || opts.TailLines != nil ||
```

Comment:
Hold on.  We know that some "query" values are files (if they contain a path-separator) but if the query is "foobar" it *might* be a file or a service - you don't know that here.  This is the "heuristic" part of the spec.  I think you simply CAN'T do this validation here.  Either ignore these confliciting params in the impl or else find a way to signal "param x was ignored" from the impl.

Reply:
I made the assumption that not having a `/` implies a file. But I am assuming you want to cover the case of `foobar` also being a file. I didn't realize the level of heuristics you wanted. I will give it a shot.

Reply:
@thockin would you be open to adding this deeper heuristic when this moves from alpha to beta? I would rather have more time to test it out given how close we are to feature freeze.

Reply:
I think @thockin is suggesting that relaxing validation _here_ would also be acceptable - is that correct?
If so I'm in favor of that.

Reply:
imo lets just assume no `no slash = no file`  ?

Reply:
I am also not fully convinced if we need this level of deeper heuristics. It is very rare to have a file called `foobar`. I am not sure the added code complexity for this corner case is worth it. This is something we should implement after we get user feedback in the alpha phase.